repos/fdbt-netex-output/src/netex-convertor/point-to-point-tickets/pointToPointTicketNetexHelpers.ts

```diff
@@ -33,17 +33,55 @@ export const hasDuplicates = (array: string[]): boolean => {
     return new Set(array).size !== array.length;
 };
 
-export const getPointToPointScheduledStopPointsList = (fareZones: FareZone[]): ScheduledStopPoints[] => {
-    // we want a copy (by value, not by reference) because we do not want to alter the fare zones array being passed in
-    const copyOfFareZones = JSON.parse(JSON.stringify(fareZones));
+export const hasDuplicatesWithinASingleFareZone = (fareZones: FareZone[]): boolean => {
+    let hasDuplicatesWithinASingleFareZone = false;
+
+    hasDuplicatesWithinASingleFareZone = fareZones.some(fz => {
+        const stopsWithinFareZone = fz.stops;
+
+        const atcoCodesOfStopsWithinFareZone = stopsWithinFareZone.map(s => s.atcoCode);
+
+        const hasDuplicateStops = hasDuplicates(atcoCodesOfStopsWithinFareZone);
+
+        return hasDuplicateStops;
+    });
+
+    return hasDuplicatesWithinASingleFareZone;
+};
 
-    let stops = getStops(copyOfFareZones);
+export const hasDuplicatesAcrossFareZones = (fareZones: FareZone[]): boolean => {
+    const stops = getStops(fareZones);
 
     const stopsAtcoCodes = stops.map(s => s.atcoCode);
 
     const hasDuplicateStops = hasDuplicates(stopsAtcoCodes);
 
-    if (hasDuplicateStops) {
+    return hasDuplicateStops;
+};
+
+export const getPointToPointScheduledStopPointsList = (fareZones: FareZone[]): ScheduledStopPoints[] => {
```

Comment:
If there a case in the data where there would be stop duplicates within a single farezone and across farezones?

In that case this code would not remove the cross farezone duplicates

Reply:
I'm not sure what you mean, can you elaborate?

When there are duplicate stops within a single fare zone, we wish to remove the duplicates. In the NeTEx they would be removed from the fareZones tag and the ScheduledStopPoints tag.

I hope that makes sense.

Reply:
Added new unit tests. :+1: